prime-router/src/main/kotlin/fhirengine/azure/FHIRProcessFunctions.kt

```diff
@@ -63,6 +63,12 @@ class FHIRFlowFunctions(
         logger.trace("Processing HL7 data for FHIR conversion.")
         try {
             val hl7messages = HL7Reader(actionLogger).getMessages(messageContent.downloadContent())
+
+            // this may need additional work in the future to ensure we are surfacing correct message(s)
+            if (actionLogger.hasErrors()) {
+                throw java.lang.IllegalArgumentException(actionLogger.errors[0].detail.message)
```

Comment:
Maybe do a `actionLogger.errors.joinToString...` or something, so we don't have to worry about having multiple errors here in the future.

Reply:
Updated.